azurerm/internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -331,6 +332,71 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				Optional: true,
 			},
 
+			"maintenance_config": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"maintenance_allowed": {
+							Type:         pluginsdk.TypeSet,
+							Optional:     true,
+							AtLeastOneOf: []string{"maintenance_config.0.maintenance_allowed", "maintenance_config.0.maintenance_not_allowed_window"},
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"day": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(containerservice.WeekDaySunday),
+											string(containerservice.WeekDayMonday),
+											string(containerservice.WeekDayTuesday),
+											string(containerservice.WeekDayWednesday),
+											string(containerservice.WeekDayThursday),
+											string(containerservice.WeekDayFriday),
+											string(containerservice.WeekDaySaturday),
+										}, false),
+									},
+
+									"hour_slots": {
```

Comment:
does slots make sense? hours seems fine
```suggestion
									"hours": {
```